packages/wallet/ui/src/components/Offer.jsx

```diff
@@ -62,9 +62,14 @@ const OfferWithoutContext = ({
     status = 'decline';
   }
 
-  const approve = () => {
+  const approve = async () => {
     setPendingOffers({ offerId: id, isPending: true });
-    E(offer.actions).accept().catch(console.error);
+    try {
+      await E(offer.actions).accept();
+    } catch (e) {
+      setPendingOffers({ offerId: id, isPending: false });
+      console.error('Failed to accept offer', e);
+    }
```

Comment:
with this `try`, returned promise will now resolve even if `accept` fails. I don't know if anything is actually paying attention to the result (there's no return type) but I think it would be safer to keep this function sync and return the failed promise.

```suggestion
    return E(offer.actions).accept().catch(e => {
      setPendingOffers({ offerId: id, isPending: false });
      console.error('Failed to accept offer', e);
    });
```

Reply:
That's better, done thanks